### PR TITLE
update the aws dependency to support to aws-sdk-v3

### PR DESCRIPTION
**Pull request by JuanLUJoanne**

### CODE REVIEW

**Review by josephecombs — COMMENTED** [review]
change to this, 3.0.1 just got released yesterday:

spec.add_dependency 'aws-sdk-resources', '~> 3'

**Comment by winebarrel** [comment]
Thanks!
Sorry for the late merge

**Comment by winebarrel** [comment]
v0.2.0.beta has been released.
However this version installs too many gems...
I will fix it in a few days

**Comment by josephecombs** [comment]
beautiful - I need this on elasticbeanstalk and that for whatever reason doesnt like pointing gem definitions at github,  so your speediness here is MUCH appreciated!

**Comment by josephecombs** [comment]
I'm getting an error on deployment:

```Message

LoadError: cannot load such file -- aws-sdk/signers/v4

Traceback

LoadError: cannot load such file -- aws-sdk/signers/v4
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in block in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 258, in load_dependency
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday_middleware-aws-signers-v4-0.2.0.beta/lib/faraday_middleware/aws_signers_v4_ext.rb", line 1, in <top (required)>
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in block in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 258, in load_dependency
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday_middleware-aws-signers-v4-0.2.0.beta/lib/faraday_middleware/request/aws_signers_v4.rb", line 2, in <top (required)>
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in block in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 258, in load_dependency
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/dependencies.rb", line 292, in require
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday_middleware-aws-signers-v4-0.2.0.beta/lib/faraday_middleware/aws_signers_v4.rb", line 7, in block in <module:FaradayMiddleware>
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday-0.13.1/lib/faraday.rb", line 213, in block in load_middleware
  File "/opt/rubies/ruby-2.3.4/lib/ruby/2.3.0/monitor.rb", line 214, in mon_synchronize
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday-0.13.1/lib/faraday.rb", line 195, in middleware_mutex
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday-0.13.1/lib/faraday.rb", line 212, in load_middleware
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday-0.13.1/lib/faraday.rb", line 186, in lookup_middleware
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday-0.13.1/lib/faraday/rack_builder.rb", line 226, in use_symbol
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday-0.13.1/lib/faraday/rack_builder.rb", line 93, in request
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick.rb", line 59, in block in client
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/client.rb", line 115, in block in initialize
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/faraday-0.13.1/lib/faraday/connection.rb", line 88, in initialize
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/http/faraday.rb", line 38, in new
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/http/faraday.rb", line 38, in __build_connection
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/base.rb", line 138, in block in __build_connections
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/base.rb", line 130, in map
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/base.rb", line 130, in __build_connections
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/base.rb", line 40, in initialize
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/client.rb", line 114, in new
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/client.rb", line 114, in initialize
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport.rb", line 26, in new
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport.rb", line 26, in new
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick.rb", line 54, in client
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick/query.rb", line 198, in execute_search
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick/logging.rb", line 13, in block in execute_search
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/notifications.rb", line 166, in block in instrument
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/notifications/instrumenter.rb", line 21, in instrument
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/activesupport-5.1.3/lib/active_support/notifications.rb", line 166, in instrument
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick/logging.rb", line 12, in execute_search
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick/query.rb", line 81, in execute
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick.rb", line 100, in search
  File "/opt/rubies/ruby-2.3.4/lib/ruby/gems/2.3.0/gems/searchkick-2.3.1/lib/searchkick/model.rb", line 29, in searchkick_search
  File "/var/app/current/lib/tasks/reindex_if_necessary.rake", line 13, in block (3 levels) in <top (required)>```

**Comment by winebarrel** [comment]
😢 
Thank you so much for your report!
I can not fix it right away...
please wait for a few days.
I will investigate the cause.

File: lib/faraday_middleware/aws_signers_v4_ext.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require 'aws-sdk-core/signers/v4'
+require 'aws-sdk/signers/v4'
```
Comment (josephecombs) [inline lib/faraday_middleware/aws_signers_v4_ext.rb:1]: should this be aws-sdk-core ?

**Comment by winebarrel** [comment]
I released v0.1.8 using aws-sdk v2.

**Comment by winebarrel** [comment]
Currently, I am developing a new gem for aws-sdk-v3.
This gem does not support aws-sdk-v3.
see https://github.com/winebarrel/faraday_middleware-aws-sigv4

Please use v0.1.8 If you use aws-sdk-v2.

**Pull request merged**